### ISSUE TITLE
[Accessibility]: Timeline steps are not numbered

### ISSUE THREAD

**Issue by Strum355** [sourcegraph/sourcegraph#35265]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33513, https://github.com/sourcegraph/sourcegraph/issues/33512

### Problem description

The timeline steps are not part of a list or have any auditory numbering, with the only indication of a list-like element approaching being the word "Timeline" at a stretch. 10:00 in the recording.

### Expected behavior

Timelines should number each step directly (hard-coded numbers) or indirectly (via list elements) to make it clear where in the timeline one is and how much there is to navigate.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
